unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -1,35 +1,67 @@
+import asyncio
+import hashlib
 import json
+import sys
+import time
+from concurrent.futures import ThreadPoolExecutor
 from dataclasses import dataclass, field
 from datetime import timedelta
 from pathlib import Path
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, Generator, List
 
 from pydantic import Field, Secret
 
-from unstructured_ingest.error import DestinationConnectionError
-from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.error import (
+    DestinationConnectionError,
+    SourceConnectionError,
+    SourceConnectionNetworkError,
+)
+from unstructured_ingest.utils.data_prep import batch_generator, flatten_dict
 from unstructured_ingest.utils.dep_check import requires_dependencies
 from unstructured_ingest.v2.interfaces import (
     AccessConfig,
     ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
     UploadContent,
     Uploader,
     UploaderConfig,
     UploadStager,
     UploadStagerConfig,
+    download_responses,
 )
 from unstructured_ingest.v2.logger import logger
 from unstructured_ingest.v2.processes.connector_registry import (
     DestinationRegistryEntry,
+    SourceRegistryEntry,
 )
 
 if TYPE_CHECKING:
-    from couchbase.cluster import Cluster
+    pass
 
 CONNECTOR_TYPE = "couchbase"
 SERVER_API_VERSION = "1"
 
 
+@requires_dependencies(["couchbase"], extras="couchbase")
+def connect_to_couchbase(connection_string: str, username: str, password: str):
```

Comment:
And add `from couchbase.cluster import Cluster` to the block under `if TYPE_CHECKING:`

```suggestion
def connect_to_couchbase(connection_string: str, username: str, password: str) -> "Cluster":
```

Reply:
This can also be a method on the `CouchbaseConnectionConfig` since it uses values that are available on it.